app/controllers/api/v1/links_controller.rb

```diff
@@ -36,6 +36,24 @@ def top_links
     render json: LinkSerializer.new(links)
   end
 
+  def update_privacy
+    user = User.find(params[:user_id])
+    link = Link.find(params[:id])
+    is_private = params[:private] == 'true'
+
+    if link.user_id == user.id
+      if link.update(private: is_private)
+        render json: { message: 'Privacy setting updated successfully' }, status: :ok
+      else
+        render json: { error: 'Failed to update privacy setting' }, status: :unprocessable_entity
+      end
+    else
+      render json: { error: 'Unauthorized to update this link' }, status: :forbidden
+    end
+  rescue ActiveRecord::RecordNotFound
+    render json: { error: 'User or Link not found' }, status: :not_found
+  end
+
```

Comment:
I like this implementation of using messages as opposed to returning full link objects